### PR TITLE
UI rework. New areas created for the user to provide examples and store them

### PR DESCRIPTION
**Pull request by ghost**
WORK IN PROGRESS.
Ticket:
https://github.com/unosquare/modgen.js/issues/7

New UI. New text areas for the user. Now it's possible to register examples (inputs and results) neccesary for GPT to understand what it needs to generate.
image
image

### CODE REVIEW

File: src/app/inicio/components.tsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+                      onChange={onChange}
+                      className="text-black h-44 w-[40rem] rounded resize-none" 
+            >
+            </textarea> <br />
+
+        </>
+    )
+}
+
+interface dataTypes{
```
Comment (AntonioGGovea) [inline src/app/inicio/components.tsx]: By convention, interface name start with a capital letter. Please change it here and in the rest of interfaces.

File: src/app/inicio/components.tsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+}
+
+//results are GPT responses exemplified by the user
+export function Result({data, value, onChange}: dataTypes){
+    const {userInputs, userResults} = userInputStore();
+    const [updateUserInputArr, examples] = userInputsArr((state:any)=> [state.updateUserInputArr, state.examples]);
+    const num = examples.length;
+
+    const updateShots = ()=>{    
+        let values = {"id":num,"input": userInputs, "results":userResults};
```
Comment (AntonioGGovea) [inline src/app/inicio/components.tsx]: you can assign examples.length directly.

File: src/app/inicio/components.tsx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+//examples are built with one user input and result
+export function Examples({send}: helpingInter){
+
+    const deleteUserInputArr = userInputsArr((state:any)=> state.deleteUserInputArr);
+
+    return(        
+        <>
+            <h1>Examples</h1>
+            <div className="h-96 w-[25rem] bg-white  text-black rounded" >
+                <Shots></Shots>
```
Comment (AntonioGGovea) [inline src/app/inicio/components.tsx]: Shots can be self closing, it does not have a child

File: src/app/inicio/page.tsx [hunk 4]
```diff
@@ -39,20 +46,24 @@ export default function Home() {
...
         </div>
 
-        <div className='px-16'>
-          <h1 className='' >GPT Response:</h1><br />
-          <p >{data}</p>     
-        </div>   
+        <div className='flex flex-col px-5 py-3 bg-gray-400 rounded-lg bg-opacity-20 backdrop-blur-lg
+                        hover:shadow-lg hover:shadow-blue-950 '>
+          <div>
+            <Model value={userInputs} onChange={updateUserInputHere}></Model>
```
Comment (AntonioGGovea) [inline src/app/inicio/page.tsx]: Model and Result can be self closing too

**Review by AntonioGGovea — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
https://github.com/unosquare/modgen.js/assets/99301366/ab8224a6-b258-461c-a1c3-b75b7d6d09c3

Demonstration.

File: src/app/inicio/components.tsx [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+        updateUserInput, 
+        updateUserResults, 
+        userResults] = userInputStore((state) => [state.userInputs, 
+                                                  state.updateUserInput, 
+                                                  state.updateUserResults, 
+                                                  state.userResults]);
+    const [updateUserInputArr] = userInputsArr((state)=> [state.updateUserInputArr]);
+
+    const updateShots = ()=>{    
+        let values:ShotsTypes = {input: userInputs, result:userResults};
```
Comment (AntonioGGovea) [inline src/app/inicio/components.tsx]: Pass the value directly to updateUserInputArr.

File: src/app/inicio/components.tsx [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+        updateUserResults("");
+    }
+    const inputRef:any = useRef(null);
+    const deleteText = ()=>{
+        if(inputRef.current != null){
+            inputRef.current.value = "";
+        }
+        if(document.getElementById('input-model-alwaysActive') != null){
+            (document.getElementById('input-model-alwaysActive') as HTMLInputElement).value = "";
+        }
```
Comment (AntonioGGovea) [inline src/app/inicio/components.tsx]: there is a better way to handle this without having to access the document or use ref by using controlled inputs.
Try reading about this, here is a page with general info as well.
https://react.dev/reference/react-dom/components/input
If you need some help or extra explanation, tell me and we can have call 🙂

**Review by AntonioGGovea — CHANGES_REQUESTED** [review]

**Review by AntonioGGovea — APPROVED** [review]

**Review by oscarfgq — APPROVED** [review]

**Pull request merged**